extended/src/main/java/apoc/vectordb/ChromaDb.java

```diff
@@ -0,0 +1,220 @@
+        config.putIfAbsent(METHOD_KEY, "DELETE");
+
+        RestAPIConfig restAPIConfig = new RestAPIConfig(config, Map.of(), Map.of());
+        return executeRequest(restAPIConfig, urlAccessChecker)
+                .map(v -> (Map<String,Object>)v)
+                .map(MapResult::new);
+    }
+
+    @Procedure("apoc.vectordb.chroma.upsert")
+    @Description("apoc.vectordb.chroma.upsert(hostOrKey, collection, vectors, $configuration) - Upserts, in the collection with the name specified in the 2nd parameter, the vectors [{id: 'id', vector: '<vectorDb>', medatada: '<metadata>'}]")
+    public Stream<MapResult> upsert(
+            @Name("hostOrKey") String hostOrKey,
+            @Name("collection") String collection,
+            @Name("vectors") List<Map<String, Object>> vectors,
+            @Name(value = "configuration", defaultValue = "{}") Map<String, Object> configuration) throws Exception {
+        String url = "%s/api/v1/collections/%s/upsert";
+        Map<String, Object> config = getVectorDbInfo(hostOrKey, collection, configuration, url);
+        
+        Map<String, String> mapKeys = Map.of("id", "ids",
+                "vector", "embeddings",
+                "metadata", "metadatas",
+                "text", "documents");
+
+        // transform to format digestible by RestAPI,
+        // that is from [{id: <id1>, vector: <vector1>,,,}, {id: <id2>, vector: <vector2>,,,}] 
+        // to {ids: [<id1>, <id2>], vectors: [<vector1>, <vector2>]}
+        Map<Object, List> additionalBodies = listOfMapToMapOfLists(mapKeys, vectors);
+        additionalBodies.compute( "ids", (k,v) -> getStringIds(v) );
+
+        RestAPIConfig restAPIConfig = new RestAPIConfig(config, Map.of(), additionalBodies);
+        return executeRequest(restAPIConfig, urlAccessChecker)
+                .map(v -> (Map<String,Object>)v)
+                .map(MapResult::new);
+    }
+
+    @Procedure(value = "apoc.vectordb.chroma.delete", mode = Mode.SCHEMA)
+    @Description("apoc.vectordb.chroma.delete(hostOrKey, collection, ids, $configuration) - Delete the vectors with the specified `ids`")
+    public Stream<ListResult> delete(@Name("hostOrKey") String hostOrKey,
+                                     @Name("collection") String collection,
+                                     @Name("ids") List<Object> ids,
+                                     @Name(value = "configuration", defaultValue = "{}") Map<String, Object> configuration) throws Exception {
+        String url = "%s/api/v1/collections/%s/delete";
+        Map<String, Object> config = getVectorDbInfo(hostOrKey, collection, configuration, url);
+
+        VectorEmbeddingConfig apiConfig = CHROMA.get().getEmbedding().fromGet(config, procedureCallContext, getStringIds(ids));
+        return executeRequest(apiConfig.getApiConfig(), urlAccessChecker)
+                .map(v -> (List) v)
+                .map(ListResult::new);
+    }
+
+    @Procedure(value = "apoc.vectordb.chroma.get", mode = Mode.SCHEMA)
```

Comment:
this might be mode=WRITE if we keep the update behavior ?

I think if we should move the write behavior into a separate method, like `queryAndUpdate` or so? or `updateGraphFromQuery` ? and keep the `query` method read-only, otherwise read-only users can't use it and accidental write behavior will be confusing.